python/cudf/cudf/core/udf/groupby_typing.py

```diff
@@ -0,0 +1,216 @@
+# Copyright (c) 2020-2023, NVIDIA CORPORATION.
+from typing import Any, Dict
+
+import numba
+import pandas as pd
+from numba import cuda, types
+from numba.core.extending import (
+    make_attribute_wrapper,
+    models,
+    register_model,
+    type_callable,
+    typeof_impl,
+)
+from numba.core.typing import signature as nb_signature
+from numba.core.typing.templates import AbstractTemplate, AttributeTemplate
+from numba.cuda.cudadecl import registry as cuda_registry
+from numba.np import numpy_support
+
+index_default_type = numpy_support.from_dtype(
+    pd.RangeIndex(0, 0).dtype
+)  # int64
```

Comment:
If pandas changed this, I _think_ what we'd get is compilation errors complaining that there wasn't a PTX function found with the correct signature for `idxmax` and `idxmin` at least. Something about the prototype not matching the source IIRC. You might need to run a UDF that tried to use it for it to fail though.

I do think that many tests would fail around the cuDF codebase outside of the context of UDFs.

Reply:
Okay. Given that we expect major breakage if pandas were to change, we could hardcode the value. I’d like to eliminate the need to construct temporary Pandas objects at import time just to know their type.

Reply:
Switched back 👍